src/rpc/handlers/LedgerEntry.cpp

```diff
@@ -29,14 +29,30 @@ doLedgerEntry(Context const& context)
     auto lgrInfo = std::get<ripple::LedgerInfo>(v);
 
     ripple::uint256 key;
-    if (request.contains(JS(index)))
+    if ((request.contains(JS(index))) && (request.contains(JS(ledger_index))))
+    {
+        if (!request.at(JS(index)).is_string())
+            return Status{Error::rpcINVALID_PARAMS, "malformedRequest"};
+
+        if (!key.parseHex(request.at(JS(index)).as_string().c_str()))
+            return Status{Error::rpcINVALID_PARAMS, "malformedRequest"};
+
+        if (!request.at(JS(ledger_index)).is_string())
+            return Status{Error::rpcINVALID_PARAMS, "malformedRequest"};
+    }
+    else if (request.contains(JS(index)))
```

Comment:
What exactly are these additions doing? We were already checking that `index` was a string below, why are we checking it again now?